src/Rule/NestedHandler.php

```diff
@@ -93,7 +93,10 @@ public function validate(mixed $value, RuleInterface $rule, ValidationContext $c
             } else {
                 $valuePathList = StringHelper::parsePath($valuePath);
                 $property = end($valuePathList);
-                $itemResult = $context->validate([$property => $validatedValue], [$property => $rules]);
+                $itemResult = $context->validate(
+                    ArrayHelper::pathExists($data, $valuePath) ? [$property => $validatedValue] : [],
```

Comment:
```suggestion
                    ArrayHelper::keyExists($data, $valuePath) ? [$property => $validatedValue] : [],
```

`$valuePath` is already an array of keys, so using `keyExists()` is a bit more efficient.

Reply:
Oh yes! Only then should check not `$valuePath`, but `$valuePathList`. Thank you, that's more correct. 👍